module-system/sov-modules-api/src/lib.rs

```diff
@@ -183,6 +194,14 @@ pub trait Spec {
         + Sync
         + FromStr<Err = anyhow::Error>;
 
+    /// The public key used for digital signatures
+    #[cfg(feature = "native")]
+    type PrivateKey: Debug
+        + Send
+        + Sync
+        + for<'a> TryFrom<&'a [u8], Error = anyhow::Error>
```

Comment:
Let's use `thiserror` instead of `anyhow`?

Reply:
I think this is the same issue we ran into before - if we want to use `thiserror`, we have to add an extra associated type to the `Spec` trait for the Error